### ISSUE TITLE
Omni sensors for SDM12 are not reporting sensor values

### ISSUE THREAD

**Issue by ptr727** [loopj/aiovantage#98]
### Checklist

- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/ludeeus/integration_blueprint/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Sensors for SDM12 (temperature, current, power) are not reporting values.

But, as soon as DC is connected, and the SDM12 is selected in the DC UI, the sensors start reporting.  
Close DC, or change focus to a different object, and sensors stop reporting again.

I am not 100% sure, but sensors for SDM12 did work when introduced, so may have stopped working some point later.

I noticed this with SDM12's, but it may apply to other omni sensors as well.

I could not confirm if temperature is reported, as DC in v4.6 that I tested with did not show temperature, at least I could not find it in the UI, have not tested with DC 4.7 yet.

### Reproduction steps

- Run `monitor_omni_sensors.py`, or add SDM12 sensors to HA dashboard.
- Observe that there are no values reported.
- Launch DC, connected to controller, focus on a SDM12 module.
- Observe that sensor values are now reported every few seconds.
- But only for the selected module.
- Change DC focus to a different module and observe values now reported for that module.
- Close DC and sensor values stop being reported.

**Comment by ptr727** [comment]
I did some more experimentation, this is what I found.

Using the OmniSensor example `python monitor_omni_sensors.py 192.168.1.3 --debug --username=administrator --password=password`:

```log
DEBUG:aiovantage.command_client.events:Sending message: LOGIN administrator password
INFO:aiovantage.command_client.events:Connected to event stream at 192.168.1.3:3010
DEBUG:aiovantage.command_client.events:Started the event stream
INFO:aiovantage.controllers:OmniSensorsController subscribed to state changes
INFO:aiovantage.controllers:OmniSensorsController initialized (36 objects)
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE STATUS ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG STATUS ON
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE STATUSEX ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG STATUSEX ON
DEBUG:aiovantage.command_client.events:Received message: R:LOGIN administrator @Vantage1
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 STATUS ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG STATUS ON
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 STATUSEX ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG STATUSEX ON
```

And the SDM12 sensor values are never updated.

Open Infusion Diagnostics Host Commands window, Click on Full, or just on Automation, and the sensor values start getting reported.

![image](https://github.com/user-attachments/assets/f64f9e59-5581-408e-baa6-983f809fe7a9)

Test in `telnet` by repeating the steps, and see the same behavior.

Modify `base.py` and `"AUTOMATION", "EVENT", "SYSTEM"`:

```python
        # Some state changes are only available from "object" status events.
        # These can be subscribed to by using "STATUSADD {vid}" or "ELLOG STATUS".
        if self.interface_status_types:
            # Subscribe to "object status" events from the Enhanced Log.
            self.event_stream.subscribe_enhanced_log(
                self._handle_event, ("STATUS", "STATUSEX",
                                     "AUTOMATION", "EVENT", "SYSTEM")
            )
```

Rerun, and now sensor values are getting reported.

```log
DEBUG:aiovantage.command_client.events:Sending message: LOGIN administrator password
INFO:aiovantage.command_client.events:Connected to event stream at 192.168.1.3:3010
DEBUG:aiovantage.command_client.events:Started the event stream
INFO:aiovantage.controllers:OmniSensorsController subscribed to state changes
INFO:aiovantage.controllers:OmniSensorsController initialized (36 objects)
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE STATUS ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG STATUS ON
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE STATUSEX ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG STATUSEX ON
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE AUTOMATION ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG AUTOMATION ON
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE EVENT ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG EVENT ON
DEBUG:aiovantage.command_client.events:Sending message: ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Sending message: ELENABLE SYSTEM ON
DEBUG:aiovantage.command_client.events:Sending message: ELLOG SYSTEM ON
DEBUG:aiovantage.command_client.events:Received message: R:LOGIN administrator password
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 STATUS ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG STATUS ON
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 STATUSEX ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG STATUSEX ON
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 AUTOMATION ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG AUTOMATION ON
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 EVENT ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG EVENT ON
DEBUG:aiovantage.command_client.events:Received message: R:ELAGG 1 ON
DEBUG:aiovantage.command_client.events:Received message: R:ELENABLE 1 SYSTEM ON
DEBUG:aiovantage.command_client.events:Received message: R:ELLOG SYSTEM ON
DEBUG:aiovantage.command_client.events:Received message: EL: 473 475 0 1 0
DEBUG:aiovantage.command_client.events:Received message: EL: 475 107 Load.GetLevel 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 473 Button.GetState 0
DEBUG:aiovantage.command_client.events:Received message: EL: 475 Task.IsRunning 1
DEBUG:aiovantage.command_client.events:Received message: EL: 475 117 Load.GetLevel 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 475 112 Load.GetLevel 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2 Controller Query 475 Complete: 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 24 Temperature.GetValue 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 25 PowerSensor.GetPower 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 26 PowerSensor.GetPower 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 27 PowerSensor.GetPower 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 28 PowerSensor.GetPower 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 29 CurrentSensor.GetCurrent 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 30 CurrentSensor.GetCurrent 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 31 CurrentSensor.GetCurrent 0 0
DEBUG:aiovantage.command_client.events:Received message: EL: 2147516426 32 CurrentSensor.GetCurrent 0 0
```

By the design of the code it looks like OmniSensor may need to add `AUTOMATION` to subscribe to events, but I am not certain enough to do a PR yet.

From the screenshot you can see how data started working after running the example app:
![image](https://github.com/user-attachments/assets/d182935b-3316-4a1f-b56e-41466872b372)

**Comment by ptr727** [comment]
With the change to add `AUTOMATION`, and running the omni test app, and HA, I now see periodic EL's for sensors, but the reported values never change. E.g. in my case all but two are 0, and the two non-zero values never change.

I `"AUTOMATION", "EVENT", "SYSTEM"`, ran the test app, logged to file, connected DC, selected the SDM module, and observed that DC seems to be polling HW values, as expected the values reported by HA reflected the polled values. I selected a SDM, power cycled some loads, observed values changing as expected. Disconnected DC, and previous values kept being reported, even after powering off loads.

I am not sure if the log captures all commands sent from other clients, else some MITM network traffic snooping between DC and the controller may be required?
 
Log attached.
[omni.log](https://github.com/user-attachments/files/17575153/omni.log)

**Comment by ptr727** [comment]
> Fixed in [0.15.0](https://github.com/loopj/aiovantage/releases/tag/0.15.0)

Or 0.16 where the cache=false was added?

**Comment by loopj** [comment]
0.16! good catch
